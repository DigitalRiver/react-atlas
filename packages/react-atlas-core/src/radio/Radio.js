import React, { Component, PropTypes } from 'react';
import { classNames } from '../utils';
import themeable from 'react-themeable';

const Radio = ({className, disabled, inline, name, label, ...props}) => {
  const theme = themeable(props.theme);
  const labelClasses = classNames({
    block: !inline,
    inline,
    disabled
  }, className);

  return (
    <label {...theme(1, ...labelClasses)}>
      <input
        {...props}
        disabled={disabled}
        name={name}
        type="radio"
      />
      {label ? <span> {label}</span> : null}
    </label>
  );
};

Radio.propTypes = {
  defaultChecked: PropTypes.bool,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  inline: PropTypes.bool,
  label: PropTypes.string,
  name: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onClick: PropTypes.func,
  onFocus: PropTypes.func,
  value: PropTypes.any
};

Radio.defaultProps = {
  className: '',
  disabled: false,
  inline: false
};

Radio.styleguide = {
  category: 'Form Components',
  index: '3.8',
  example: `
<section>
  <h5>Individual Radio Buttons</h5>
  <p>Not aware of each other, use RadioGroup for that.</p>
  <Radio label="Checked Radio" value="checkedRadio" defaultChecked />
  <Radio label="Disabled Radio" value="disabledRadio" disabled/>
  <Radio label="Inlined 1" value="inlined1" className="someClassIMadeUp" inline />
  <Radio label="Inlined 2" value="inlined2" inline />
</section>
`
};

export default Radio;
