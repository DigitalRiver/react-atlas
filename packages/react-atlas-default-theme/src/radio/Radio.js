import { Radio } from "react-atlas-core";
import style from "./radio.css";
import createComponent from "../createComponent";

export default createComponent(Radio, style);
