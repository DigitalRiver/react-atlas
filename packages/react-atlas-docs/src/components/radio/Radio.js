import React, { PropTypes } from "react";
import * as Atlas from 'react-atlas';

export default class Radio extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <Atlas.Radio {...this.props}></Atlas.Radio>
    )
  }
}