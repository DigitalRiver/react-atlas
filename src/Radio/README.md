The Radio component is a controlled component that must be used within the RadioGroup component. For more examples view the RadioGroup examples.

###### Default Radio:

    <RadioGroup name="radioTest">
			<Radio label="Option 1" value="first"/>
			<Radio label="Option 2" value="second"/>
		</RadioGroup>

###### Disabled Radio:

    <RadioGroup name="disabledTest">
			<Radio label="Option 1" value="first"/>
			<Radio label="Option 2" value="second" disabled/>
			<Radio label="Option 3" value="third"/>
		</RadioGroup>

###### Hidden Radio:

    <RadioGroup name="hiddenTest">
			<Radio label="Option 1" value="first"/>
			<Radio label="Option 2" value="second" hidden/>
			<Radio label="Option 3" value="third"/>
		</RadioGroup>

###### Radio with left-positioned label:

    <RadioGroup name="leftLabelTest">
			<Radio label="Option 1" labelPosition="left" value="first"/>
			<Radio label="Option 2" value="second"/>
		</RadioGroup>

###### Custom onClick method:

    <RadioGroup name="onClickTest">
			<Radio label="Custom onClick" value="first" onClick={ (e, data) => { console.log(data.index, data.value) } }/>
			<Radio label="Normal Radio" value="second"/>
		</RadioGroup>

###### Custom onChange method:

    <RadioGroup name="onChangeTest">
			<Radio label="Custom onChange" value="first" onChange={ (data) => { console.log(data.checked, data.value) } }/>
			<Radio label="Normal Radio" value="second"/>
		</RadioGroup>

###### Custom function before change:

    <RadioGroup name="onBeforeChangeTest">
			<Radio 
				label="Custom onBeforeChange"
				value="first" 
				onBeforeChange={ function(e, data){ if (!data.checked) { let accept = confirm("Do you want to check this?"); return accept; } return true; } }/>
			<Radio label="Normal Radio" value="second"/>
		</RadioGroup>
