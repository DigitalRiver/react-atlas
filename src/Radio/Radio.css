@import '../styles/variables.css';

.disabled {
    composes: default-text from '../styles.css';
}

.relative {
    position: relative;
}

.padding {
    padding-bottom: 5px;
}

.input {
    composes: relative;
    top: 1px;
}

.radio {
    position: absolute;
    top: 0;
    left: 0;
    height: 16px;
    width: 16px;
    display: inline-block;
    composes: cursor-pointer from '../styles.css';
    composes: default-border from '../styles.css';
    composes: circle from '../styles.css';
    background: -webkit-linear-gradient(#fff, #e6e6e6); /* For Safari 5.1 to 6.0 */
    background: -o-linear-gradient(#fff, #e6e6e6); /* For Opera 11.1 to 12.0 */
    background: -moz-linear-gradient(#fff, #e6e6e6); /* For Firefox 3.6 to 15 */
    background: linear-gradient(#fff, #e6e6e6); /* Standard syntax */
}

.disabled {
    opacity: 0.65;
    composes: cursor-not-allowed from '../styles.css';
    pointer-events: none;
}

.hidden {
    display: none;
    visibility: hidden;
}

.checked {
    composes: radio;
    composes: white-inset-shadow from '../styles.css';
    background: var(--primary);
}

.focus {
    box-shadow: 0 0 6px rgba(35, 173, 255, 1);
}

.checkmark {
    composes: relative;
    top: 0px;
}

.error {
    border: solid 1px var(--error);
}

.label {
    composes: default-font from '../styles.css';
    cursor: pointer;
    white-space: nowrap;
    padding-right: 5px;
    margin-left: 5px;
}

.label_left {
    composes: label;
    padding-right: 5px;
    margin-left: 0;
    margin-right: 5px;
}

.inline_block {
    display: inline-block;
}

.radio_padding {
    padding: 7px 10px 0 0;
}

.float_left {
    float: left;
    padding-right: 3px;
}

.float_right {
    float: right;
    padding-left: 3px;
    position: relative;
}

.input_style {
    opacity: 0; /*this needs to stay here to avoid display issues in IE*/
}