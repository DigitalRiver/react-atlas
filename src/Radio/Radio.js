import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Radio.css";
import blacklist from "blacklist";

export class Radio extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    /* Since Radio is controlled, the only changes that can occur need to come from property updates */
    if (this.props.onChange && nextProps.checked !== this.props.checked) {
      this.props.onChange({
        "checked": nextProps.checked,
        "value": this.props.value
      });
    }
  }

  // Handles new radio clicks and sets value and checked status of hidden input
  _clickHandler = e => {
    e.persist();
    if (this.props.disabled) {
      return false;
    }

    if (this.props.onClick) {
      this.props.onClick(e, {
        "index": this.props.index,
        "value": this.props.value
      });
    }

    if (
      typeof this.props.onBeforeChange === "undefined" ||
      this.props.onBeforeChange(e, {
        "checked": this.props.checked,
        "value": this.props.value
      })
    ) {
      this.props.groupSetChecked(e, this.props.index);
    }
  };

  // Added to remove console warning for controlled/uncontrolled component
  _handleChange = () => {
    return false;
  };

  render() {
    const {
      checked,
      className,
      disabled,
      hidden,
      id,
      inline,
      label,
      labelPosition,
      style,
      title,
      ...others
    } = this.props;

    // Declaring the following variables so they don't get passed to the DOM element through the prop spread.
    const othersFiltered = blacklist(
      others,
      "groupSetChecked",
      "onBeforeChange"
    );

    /* Classes and styles setup */
    let wrapperStyles = cx(
      {
        "inline_block": inline,
        "hidden": hidden
      },
      "radio_padding"
    );

    let labelStyle = cx(
      {
        "label_left": labelPosition === "left"
      },
      "label"
    );

    let radioDisplay = labelPosition === "left" ? "float_right" : "float_left";

    let labelTitle = title || label;

    let radioClass = cx({
      "radio": true,
      "checked": checked
    });

    return (
      <div style={style} onClick={this._clickHandler} styleName={wrapperStyles}>
        <div
          styleName={cx({ disabled }, "inline_block", "relative", "padding")}
        >
          {label && 
            <label
              styleName={labelStyle}
              title={labelTitle}
              className={cx(className)}
              htmlFor={id}
            >
              {label}
            </label>
          }
          <div styleName={radioDisplay}>
            <input
              {...othersFiltered}
              type="radio"
              checked={checked}
              disabled={disabled}
              hidden={hidden}
              id={id}
              onChange={this._handleChange}
              styleName={cx("input_style")}
            />
            <div styleName={radioClass}>
              {checked && <div styleName={"checkmark"} />}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Radio.propTypes = {
  /**
   * When true, Radio will be checked.
   * @examples '<Radio checked/>'
   */
  "checked": PropTypes.bool,

  /** An object, array, or string of CSS classes to apply to Radio.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * When true, Radio will be disabled.
   * @examples '<Radio disabled/>'
   */
  "disabled": PropTypes.bool,

  /**
   * When true, Radio will be hidden.
   * @examples '<Radio hidden/>'
   */
  "hidden": PropTypes.bool,
  /** Will set the html "id" property on the Radio. */
  "id": PropTypes.string,
  /** Represents the index of the Radio within the parent RadioGroup. */
  "index": PropTypes.number,
  /**
   * When true, Radio will display inline.
   * @examples '<Radio inline/>'
   */
  "inline": PropTypes.bool,

  /**
   * Function that will be used as a callback to send to RadioGroup component.
   * @examples '<Radio groupSetChecked={...}/>'
   */
  "groupSetChecked": PropTypes.func,

  /**
   * Defines the text that will be displayed for Radio label.
   * @examples '<Radio label="test"/>'
   */
  "label": PropTypes.string,

  /**
   * Defines the label position.
   * @examples '<Radio labelPosition="top"/>'
   */
  "labelPosition": PropTypes.string,

  /**
   * Define a name for the Radio. Not necessary if using RadioGroup component as wrapper.
   * @examples '<Radio name="test"/>'
   */
  "name": PropTypes.string,

  /**
   * Function that will be executed before onChange event occurs.
   * @examples '<Radio onBeforeChange={onBeforeChangeHandler}/>'
   */
  "onBeforeChange": PropTypes.func,

  /**
   * Function that will be executed when onChange event occurs.
   * @examples '<Radio onChange={onChangeHandler}/>'
   */
  "onChange": PropTypes.func,

  /**
   * Function that will be executed when onClick event occurs.
   * @examples '<Radio onClick={onClickHandler}/>'
   */
  "onClick": PropTypes.func,

  /**
   * Pass inline styling here.
   */
  "style": PropTypes.object,

  /**
   * Defines the title attribute.
   * @examples '<Radio title="Test"/>'
   */
  "title": PropTypes.string,

  /**
   * Defines a value for the radio button.
   * @examples '<Radio value="100"/>'
   */
  "value": PropTypes.any
};

Radio.defaultProps = {
  "className": "",
  "disabled": false,
  "hidden": false,
  "inline": false
};

export default CSSModules(Radio, styles, { "allowMultiple": true });
