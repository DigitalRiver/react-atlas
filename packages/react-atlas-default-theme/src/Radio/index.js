import * as RadioStyle from "./radio.css";
export { RadioStyle };
