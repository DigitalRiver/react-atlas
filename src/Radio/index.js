import Radio from "./Radio.js";
export { Radio };
