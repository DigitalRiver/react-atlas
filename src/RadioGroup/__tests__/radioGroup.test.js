import React from "react";
import { mount } from "enzyme";
import { Radio } from "../../Radio/index";
import { RadioGroup } from "../index";

import renderer from "react-test-renderer";

describe("Test RadioGroup component render", () => {
  it("Render correctly", () => {
    const tree = renderer
      .create(
        <RadioGroup inline name="test">
          <Radio label="Option 1" value="first" />
          <Radio label="Option 2" value="second" />
          <Radio label="Option 3" value="third" />
          <Radio label="Option 4" value="fourth" />
          <Radio label="Option 5" value="fifth" />
          <Radio label="Option 6" value="sixth" />
        </RadioGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Testing radio component", () => {
  it("RadioGroup - Basic test", function() {
    mount(
      <RadioGroup inline name="test" title="Title" classname="Class">
        <Radio label="Option 1" value="first" />
        <Radio label="Option 2" value="second" />
      </RadioGroup>
    );
  });

  it("RadioGroup - Change selection test", function() {
    const result = mount(
      <RadioGroup inline name="test">
        <Radio label="Option 1" value="first" />
        <Radio label="Option 2" value="second" />
        <Radio label="Option 3" value="third" />
        <Radio label="Option 4" value="fourth" />
        <Radio label="Option 5" value="fifth" />
        <Radio label="Option 6" value="sixth" />
      </RadioGroup>
    );
    expect(result.state().checkedRadio).toEqual(0);

    result
      .find(Radio)
      .last()
      .simulate("click");
    expect(result.state().checkedRadio).toEqual(5);

    result
      .find(Radio)
      .first()
      .simulate("click");
    expect(result.state().checkedRadio).toEqual(0);
  });

  it("RadioGroup - Change props ", function() {
    const result = mount(
      <RadioGroup inline name="test">
        <Radio label="Option 1" value="first" />
        <Radio label="Option 2" value="second" />
        <Radio label="Option 3" value="third" />
        <Radio label="Option 4" value="fourth" />
        <Radio label="Option 5" value="fifth" />
        <Radio label="Option 6" value="sixth" />
      </RadioGroup>
    );
    result.setProps({ "name": "Name" });

    result
      .find(Radio)
      .last()
      .simulate("click");
    expect(result.state().checkedRadio).toEqual(5);

    result.setProps({ "selectedIndex": 1 });
    expect(result.state().checkedRadio).toEqual(1);
  });
});
