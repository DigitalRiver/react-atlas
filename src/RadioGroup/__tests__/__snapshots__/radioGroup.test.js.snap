// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RadioGroup component render Render correctly 1`] = `
<div
  className="inline"
  style={undefined}
>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 1"
      >
        Option 1
      </label>
      <div
        className="float_left"
      >
        <input
          checked={true}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={0}
          name="test"
          onChange={[Function]}
          type="radio"
          value="first"
        />
        <div
          className="radio checked"
        >
          <div
            className="checkmark"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 2"
      >
        Option 2
      </label>
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={1}
          name="test"
          onChange={[Function]}
          type="radio"
          value="second"
        />
        <div
          className="radio"
        />
      </div>
    </div>
  </div>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 3"
      >
        Option 3
      </label>
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={2}
          name="test"
          onChange={[Function]}
          type="radio"
          value="third"
        />
        <div
          className="radio"
        />
      </div>
    </div>
  </div>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 4"
      >
        Option 4
      </label>
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={3}
          name="test"
          onChange={[Function]}
          type="radio"
          value="fourth"
        />
        <div
          className="radio"
        />
      </div>
    </div>
  </div>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 5"
      >
        Option 5
      </label>
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={4}
          name="test"
          onChange={[Function]}
          type="radio"
          value="fifth"
        />
        <div
          className="radio"
        />
      </div>
    </div>
  </div>
  <div
    className="radio_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <label
        className="label"
        htmlFor={undefined}
        title="Option 6"
      >
        Option 6
      </label>
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="input_style"
          disabled={false}
          hidden={false}
          id={undefined}
          index={5}
          name="test"
          onChange={[Function]}
          type="radio"
          value="sixth"
        />
        <div
          className="radio"
        />
      </div>
    </div>
  </div>
</div>
`;
